src/main/entity/Promotion.ts

```diff
@@ -82,6 +81,16 @@ export class Promotion {
   })
   discount: Discount;
 
+  /*
+   * OneToOne bidirectional relationship between Promotion and Restaurant
+   * Each promotion can have a single restaurant
+   * */
+  @OneToOne(() => Restaurant, (restaurant) => restaurant.promotion, {
+    cascade: true,
+    nullable: false,
+  })
+  restaurant: Restaurant;
```

Comment:
Omg okay that gave me a new idea,
I can do a `ManyToOne` relationship with
```
Restaurant
   id:
   placeId:
   lat:
   lon:
```

When creating a new promotion, we try to find a restaurant with the same placeId first, otherwise we create a new restaurant (thus new placeId). 

I think I'll need to test the cascade deletes such that:
If many promotions reference the same promotion, if you delete a promotion it won't delete the restaurant. But if you delete all the promotions, it would delete the restaurant...

Reply:
Ohhh yeah that sounds like a good idea, I like it! And then in the case that the placeId is NOT_FOUND, we leave it as empty string again? So then `id` would be the primary key and `placeId` would be non-unique?

And ya that's a good point, I can't think of what it would be off the top of my head... maybe to not delete on cascade? Yeah maybe double-check that 😅